src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
What is you general feeling about this trait?

I'm not extremely used to OpenGL, but fom what I've seen in glutin/glium, it looks like it pretty much covers the fundamental abstraction of "an OpenGL canvas to draw on".

On a nitpick, we should not directly expose glutin's types, but rather expose our own and do the translation in the impl of the backend (possibly via a `From<_>` impl).